homeassistant/components/zwave_js/__init__.py

```diff
@@ -58,19 +58,24 @@ def register_node_in_dev_reg(
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Z-Wave JS from a config entry."""
     client = ZwaveClient(entry.data[CONF_URL], async_get_clientsession(hass))
+    connected = asyncio.Event()
     initialized = asyncio.Event()
     dev_reg = await device_registry.async_get_registry(hass)
 
     async def async_on_connect() -> None:
         """Handle websocket is (re)connected."""
         LOGGER.info("Connected to Zwave JS Server")
+        connected.set()
         if initialized.is_set():
             # update entity availability
             async_dispatcher_send(hass, f"{DOMAIN}_{entry.entry_id}_connection_state")
 
+        initialized.clear()
```

Comment:
I'm not following completely how these cases combine with my understanding of the lifetime events of the server.

In a normal flow when first starting the integration the `connect` event happens before the `initialized`, I'm assuming. So then it makes sense to first wait for the `connect` and then for the `initialized`. But above we have a check for `initialized.is_set`. Is this to cover the reconnect case, ie when the driver is already initialized, but we lost the connection and then reconnected? Why is it important to clear the `initialized` event then?

Reply:
So there are two life cycles.

The life cycle of the client and the life cycle of the connection. 

The connection is "connected", "initialized" (received state and listening for events), "disconnected". This is what the events from client are fired for.

For the client, "initialized" has a slightly different meaning. Best described in this issue https://github.com/home-assistant-libs/zwave-js-server-python/issues/66 . 

My idea of clearing it is that we know the client is not initialized if we just got the "connect event. We don't currently use this event but might in the future. I just wanted to make sure that the event represented the correct state of the client.